slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1256,25 +1256,44 @@ int DetectorImpl::acquire() {
     return OK;
 }
 
-void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
-    // handle Mythen3 synchronization
-    if (shm()->detType == defs::MYTHEN3 && size() > 1) {
-        std::vector<int> master;
-        std::vector<int> slaves;
-        if (positions.empty() ||
-            (positions.size() == 1 && positions[0] == -1)) {
-            positions.resize(modules.size());
-            std::iota(begin(positions), end(positions), 0);
+void DetectorImpl::getMasterSlaveList(std::vector<int> positions,
+                                      std::vector<int> &master,
+                                      std::vector<int> &slaves) {
+    // expand positions list
+    if (positions.empty() || (positions.size() == 1 && positions[0] == -1)) {
+        positions.resize(modules.size());
+        std::iota(begin(positions), end(positions), 0);
+    }
+    // could be all slaves in positions
+    slaves.reserve(positions.size());
+    auto is_master = Parallel(&Module::isMaster, positions);
+    for (size_t i : positions) {
+        if (is_master[i])
+            master.push_back(i);
+        else
+            slaves.push_back(i);
+    }
+}
+
+void DetectorImpl::startAcquisition(const bool blocking, Positions pos) {
+    bool handleSynchronization = false;
+    // multi module m3 or multi module sync enabled jungfrau
+    if (size() > 1) {
+        if (shm()->detType == defs::MYTHEN3) {
+            handleSynchronization = true;
         }
-        // could be all slaves in positions
-        slaves.reserve(positions.size());
-        auto is_master = Parallel(&Module::isMaster, positions);
-        for (size_t i : positions) {
-            if (is_master[i])
-                master.push_back(i);
-            else
-                slaves.push_back(i);
+        if ((shm()->detType == defs::JUNGFRAU) &&
+            (Parallel(&Module::getSynchronizationFromStopServer, pos)
+                 .tsquash("Inconsistent synchronization among modules"))) {
+            handleSynchronization = true;
         }
+    }
+
+    // slaves first
+    if (handleSynchronization) {
```

Comment:
I think handleSynchronization could be factored out as a function since the check is the same every time.

Reply:
Yes. Will do

Reply:
Looks good. Maybe in the future if we end up with many calls like this we can add a SynchronisedParallel call.